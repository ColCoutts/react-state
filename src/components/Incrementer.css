.incrementer {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 25vw;

  .number {
    color: lightcoral;
    cursor: pointer;

    &:hover {
      color: blue;
      transform: translateY(2px);
    }
  }
}
