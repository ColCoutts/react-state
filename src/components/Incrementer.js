import React, { PureComponent } from 'react';
import styles from './Incrementer.css';

export default class Incrementer extends PureComponent {
  state = {
    click: 0
  }

  handleClick = () => {
    this.setState(state => {
      return {
        click: state.click + 1
      };
    });
  }

  render() {
    return (
      <div className={styles.incrementer}>
        <div className={styles.number}>{this.state.click}</div>
        <button onClick={this.handleClick}>UPDATE NUMBUHR</button>
      </div>
    );
  }

}
