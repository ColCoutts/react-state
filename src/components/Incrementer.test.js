import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('incrementer component', () => {
  it('renders incrementer component', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('increments on count button', done => {
    const wrapper = shallow(<Incrementer />);

    wrapper.find('button').at(0).simulate('click');

    expect(wrapper.state('click')).toEqual(1);
    done();
  });
});
