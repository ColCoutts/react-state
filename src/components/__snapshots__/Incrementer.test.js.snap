// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrementer component renders incrementer component 1`] = `
<div
  className="incrementer"
>
  <div
    className="number"
  >
    0
  </div>
  <button
    onClick={[Function]}
  >
    UPDATE NUMBUHR
  </button>
</div>
`;
